samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,479 @@
+//DaSiamRPN tracker.
+//Original paper: https://arxiv.org/abs/1808.06048
+//Link to original repo: https://github.com/foolwood/DaSiamRPN
+//Links to onnx models:
+//network:     https://www.dropbox.com/s/rr1lk9355vzolqv/dasiamrpn_model.onnx?dl=0
+//kernel_r1:   https://www.dropbox.com/s/999cqx5zrfi7w4p/dasiamrpn_kernel_r1.onnx?dl=0
+//kernel_cls1: https://www.dropbox.com/s/qvmtszx5h339a0w/dasiamrpn_kernel_cls1.onnx?dl=0
+// sample data: https://drive.google.com/file/d/1YWOi8ma1ObT6oJRMihbk-L-9boNjCPjW/view?usp=sharing
+//
+#include <iostream>
+#include <cmath>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+using namespace cv;
+using namespace cv::dnn;
+
+const char *keys =
+        "{ help     h   | | Print help message. }"
+        "{ input    i   | ./bag | Full path to input images folder. Sample data: https://drive.google.com/file/d/1YWOi8ma1ObT6oJRMihbk-L-9boNjCPjW/view?usp=sharing }"
+        "{ net          |./dasiamrpn_model.onnx | Full path to onnx model of net.}"
+        "{ kernel_cls1  |./dasiamrpn_kernel_cls1.onnx | Full path to onnx model of net.}"
+        "{ kernel_r1    |./dasiamrpn_kernel_r1.onnx | Full path to onnx model of net.}";;
+
+typedef struct Box
+{
+    float x;
+    float y;
+    float w;
+    float h;
+} Box;
+
+// Initial parameters of the model
+typedef struct trackerConfig
+{
+    float windowInfluence = 0.43;
+    float lr = 0.4;
+    int scale = 8;
+    bool swapRB = false;
+    int totalStride = 8;
+    float penaltyK = 0.055;
+    int exemplarSize = 127;
+    int instanceSize = 271;
+    float contextAmount = 0.5;
+    std::vector<float> ratios = {0.33, 0.5, 1, 2, 3};
+    int anchorNum = ratios.size();
+    Mat anchors;
+    Mat windows;
+    Scalar avgChans;
+    Size imgSize = {0,0};
+    Box targetBox = {0,0,0,0};
+    int scoreSize = (instanceSize-exemplarSize)/totalStride+1;
+
+    std::vector<float> initBox= {334.02, 128.36, 438.19, 188.78, 396.39, 260.83, 292.23, 200.41};
```

Comment:
As I see here, you are trying to reproduce the original version of the tracker from https://github.com/foolwood/DaSiamRPN/ .
Maybe it is gonna be better to use as an example our version from 
https://github.com/opencv/opencv/blob/master/samples/dnn/dasiamrpn_tracker.py ?

Reply:
Thank you for pointing this out, the new code has added the mouse to select the initial box.